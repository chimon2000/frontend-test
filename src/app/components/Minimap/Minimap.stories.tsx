import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { action } from "@storybook/addon-actions";
import { withKnobs, text, select } from "@storybook/addon-knobs";

import { Minimap, MapCoordinateProps } from "./Minimap";

const mockCoords: MapCoordinateProps = {
  latitude: 37.33,
  longitude: -121.89
};

const stories = storiesOf("Components", module);
stories.add(
  "Minimap",
  withInfo({ inline: true })(() => (
    <Minimap
      data={mockCoords}
      popup={text("Popup Marker", "Hello, from Map!")}
      size={select("Size", ["small", "normal", "large"], "normal")}
      label={text("Optional Label", "123 My Address, Somewhere, CA, 12354")}
    />
  ))
);
