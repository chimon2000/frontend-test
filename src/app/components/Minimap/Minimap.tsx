import * as React from "react";
import * as Leaflet from "leaflet";
import { Map, Marker, Popup, TileLayer } from "react-leaflet";
import { joinClassList } from "../../helpers";

Leaflet.Icon.Default.imagePath =
  "//cdnjs.cloudflare.com/ajax/libs/leaflet/1.0.0/images/";

export interface MapCoordinateProps {
  latitude: number;
  longitude: number;
}

export interface MinimapProps {
  /**
   * Must include latitude and longitude as numbers.
   */
  data: MapCoordinateProps;
  /**
   * Optionally show a marker with popup text over the location target
   */
  popup?: string;
  className?: string;
  /**
   * Adjusts the size (height) of the map.
   */
  size?: "small" | "normal" | "large";
  /**
   * Optionally include a text label below the Map.
   */
  label?: string;
}

export const Minimap: React.FunctionComponent<MinimapProps> = ({
  data,
  popup,
  className,
  size = "normal",
  label
}) => {
  const { latitude, longitude } = data;
  const zoom: number = 13;
  const classes = [`minimap`, `${className || ""}`, `minimap__${size}`];

  return (
    <div className={joinClassList(classes)}>
      <div className="minimap--module-wrapper">
        <Map
          center={[latitude, longitude]}
          zoom={zoom}
          className="minimap--module"
        >
          <TileLayer
            attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
            url="https://{s}.tile.osm.org/{z}/{x}/{y}.png"
          />
          {popup ? (
            <Marker
              position={[latitude, longitude]}
              className="minimap--marker"
            >
              <Popup className="minimap--popup">
                <span className="minimap--popup-label">{popup}</span>
              </Popup>
            </Marker>
          ) : null}
        </Map>
      </div>
      {label ? <div className="minimap--label">{label}</div> : null}
    </div>
  );
};

export default Minimap;
