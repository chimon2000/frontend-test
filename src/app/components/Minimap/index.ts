export {
  Minimap as default,
  MapCoordinateProps,
  MinimapProps
} from "./Minimap";
