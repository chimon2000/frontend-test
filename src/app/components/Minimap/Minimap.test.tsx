import * as React from "react";
import { shallow } from "enzyme";

import Minimap, { MapCoordinateProps } from "./Minimap";

const mockCoords: MapCoordinateProps = {
  latitude: 37.33,
  longitude: -121.89
};

describe("<Minimap />", () => {
  it("doesn't blow up", () => {
    const wrapper = shallow(<Minimap data={mockCoords} />);

    expect(wrapper).toMatchSnapshot();
  });
});
