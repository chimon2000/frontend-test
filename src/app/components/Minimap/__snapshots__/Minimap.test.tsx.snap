// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Minimap /> doesn't blow up 1`] = `
<div
  className="minimap minimap__normal"
>
  <div
    className="minimap--module-wrapper"
  >
    <Map
      center={
        Array [
          37.33,
          -121.89,
        ]
      }
      className="minimap--module"
      zoom={13}
    >
      <ForwardRef(Leaflet(TileLayer))
        attribution="Â© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
        url="https://{s}.tile.osm.org/{z}/{x}/{y}.png"
      />
    </Map>
  </div>
</div>
`;
